hi-ml-cpath/testhisto/conftest.py

```diff
@@ -73,3 +75,24 @@ def tmp_path_to_pathmnist_dataset(tmp_path_factory: pytest.TempPathFactory) -> G
     download_azure_dataset(tmp_dir, dataset_id=MockHistoDataType.PATHMNIST.value)
     yield tmp_dir
     shutil.rmtree(tmp_dir)
+
+
+@pytest.fixture(scope="session")
```

Comment:
How much time does that cost? Are session scoped fixtures created under all circumstances, even if they are not used (for example, because we only run a single test that does not use the fixture)?

Reply:
Not much, 
pytest takes 3m 43s and gpu tests 5m 54s 
vs 3m 39s / 5m 20s (before this change) 
Session scoped fixtures are not executed unless supplied as an argument to a test_* function
https://stackoverflow.com/questions/26899001/pytest-how-do-i-use-global-session-wide-fixtures

I think you might be referring to `autouse=True` ? 
https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request

Reply:
I was more thinking of a case where a fixture is defined, but not actually used. Would it still execute something? I just verified that this is not the case.